ethereum/contracts/interfaces/relayer/IWormholeRelayer.sol

```diff
@@ -1,376 +1,579 @@
+   * If status is RECEIVER_FAILURE, this is 132 bytes of the return data (with revert reason information).
+   * If status is FORWARD_REQUEST_FAILURE, this is also the return data,
+   * which is specifically an error ForwardNotSufficientlyFunded(uint256 amountOfFunds, uint256 amountOfFundsNeeded)
+   * @custom:member refundStatus Result of the refund. REFUND_SUCCESS or REFUND_FAIL are for refunds where targetChainId=refundChainId; the others are for targetChainId!=refundChainId,
+   * where a cross chain refund is necessary
+   * @custom:member overridesInfo // empty if not a override, else is the encoded DeliveryOverride struct
+   */
+  event Delivery(
+    address indexed recipientContract,
+    uint16 indexed sourceChainId,
+    uint64 indexed sequence,
+    bytes32 deliveryVaaHash,
+    DeliveryStatus status,
+    uint32 gasUsed,
+    RefundStatus refundStatus,
+    bytes additionalStatusInfo,
+    bytes overridesInfo
+  );
+
+  /**
+   * @notice The relay provider calls 'deliver' to relay messages as described by one delivery instruction
+   *
+   * The instruction specifies the target chain (must be this chain), target address, refund address, maximum refund (in this chain's currency),
+   * receiver value (in this chain's currency), upper bound on gas, and the permissioned address allowed to execute this instruction
+   *
+   * The relay provider must pass in the signed wormhole messages (VAAs) from the source chain
+   * as well as the signed wormhole message with the delivery instructions (the delivery VAA)
+   * as well as identify which of the many instructions in the multichainSend container is meant to be executed
+   *
+   * The messages will be relayed to the target address (with the specified gas limit and receiver value) iff the following checks are met:
+   * - the delivery VAA has a valid signature
+   * - the delivery VAA's emitter is one of these CoreRelayer contracts
+   * - the delivery instruction container in the delivery VAA was fully funded
+   * - msg.sender is the permissioned address allowed to execute this instruction
+   * - the relay provider passed in at least [(one wormhole message fee) + instruction.maximumRefundTarget + instruction.receiverValueTarget] of this chain's currency as msg.value
+   * - the instruction's target chain is this chain
+   * - the relayed signed VAAs match the descriptions in container.messages (the VAA hashes match, or the emitter address, sequence number pair matches, depending on the description given)
+   *
+   * @param targetParams struct containing the signed wormhole messages and encoded delivery instruction container (and other information)
+   */
+  function deliver(TargetDeliveryParameters memory targetParams) external payable;
+}
+
+interface IWormholeRelayer is IWormholeRelayerSend, IWormholeRelayerDelivery {}
```

Comment:
Users will only need IWormholeRelayerSend. Do you think it is preferable to hand them this full interface rather than simply have IWormholeRelayerSend in a seperate file that users can be given? I don't have a strong opinion here but curious about the decision to put both in one file

Reply:
No, this should definitely be separated again. I didn't find the time to give the interfaces the amount of love that they need. I only restructured things and threw them together because it was easy at the time and then I didn't find the time to neatly separate them out again.

I also mentioned this on the call with @JoeHowarth just now: The errors should be grouped too so you have an idea which ones come from Send, which from Delivery and which from both and what are the circumstances when they are thrown.